portal/server/routes/contract/bond/index.js

```diff
@@ -192,13 +196,14 @@ const bondFinancialDetailsPayloadProperties = [
 
 const filterBondFinancialDetailsPayload = (body) => {
   const payload = constructPayload(body, bondFinancialDetailsPayloadProperties);
+
   if (payload.currencySameAsSupplyContractCurrency === 'true') {
-    delete payload.currency;
```

Comment:
This is because deleting the currency removes a key property where the original deal currency is stored.
This further crashes `tfm-api` upon deal submission.